compiler/injectdestructors.nim

```diff
@@ -1113,6 +1125,7 @@ proc moveOrCopy(dest, ri: PNode; c: var Con; s: var Scope, flags: set[MoveOrCopy
       # We know the result will be a stmt so we use that fact to optimize
       handleNestedTempl(ri, process, willProduceStmt = true)
     else:
+      c.checkOpError(ri, attachedAsgn)
```

Comment:
Instead it should not have been computed to be a cursor. Also, only in this one case we now get a more informative error message. Sorry, but you need to stop hacking things into random places just to make tests pass but instead need to understand the overall architecture of the compiler and look into existing logic and why it fails.

Reply:
for whatever reason it will fails, the user code want do an assignment, but have already defined `=copy` as error.